client/lib/posts/utils.js

```diff
@@ -217,6 +209,34 @@ var utils = {
 	},
 
 	/**
+	 * Returns the ID of the featured image assigned to the specified post, or
+	 * `undefined` otherwise. A utility function is useful because the format
+	 * of a post varies between the retrieve and update endpoints. When
+	 * retrieving a post, the thumbnail ID is assigned in `post_thumbnail`, but
+	 * in creating a post, the thumbnail ID is assigned to `featured_image`.
+	 *
+	 * @param  {Object} post Post object
+	 * @return {Number}      The featured image ID
+	 */
+	getFeaturedImageId: function( post ) {
+		if ( ! post ) {
+			return;
```

Comment:
Strictly speaking, this return would violate the contract set by the JS docs, should we update the return param to `Number || null`?